src/model/ScheduleFile.cpp

```diff
@@ -412,6 +412,22 @@ namespace model {
     OS_ASSERT(ok);
   }
 
+  ScheduleFile::ScheduleFile(const Model& model, const std::string& filename, int column, int rowsToSkip)
+    : ScheduleInterval(ScheduleFile::iddObjectType(), model) {
+    OS_ASSERT(getImpl<detail::ScheduleFile_Impl>());
+
+    boost::optional<ExternalFile> externalfile = ExternalFile::getExternalFile(model, filename, false);
+    OS_ASSERT(externalfile);
+
+    bool ok;
+    ok = setPointer(OS_Schedule_FileFields::ExternalFileName, externalfile->handle());
```

Comment:
getExternalFile can fail, at which point you want to remove() the object being created and throw. don't commit this as is, I left a comment in there 

```suggestion
  ScheduleFile::ScheduleFile(const Model& model, const openstudio::path& filePath, int column, int rowsToSkip)
    : ScheduleInterval(ScheduleFile::iddObjectType(), model) {
    OS_ASSERT(getImpl<detail::ScheduleFile_Impl>());

    // NOTE: if you don't end up modifying the type argument in getExternalFile, a toString(filePath) is needed
     boost::optional<ExternalFile> externalFile = ExternalFile::getExternalFile(model, filePath, false);
     if (!externalFile) {
       remove();
       LOG_AND_THROW("Couldn't find the ExternalFile at " << filePath);
    }

    bool ok = setPointer(OS_Schedule_FileFields::ExternalFileName, externalFile->handle());
```